cluster-autoscaler/cloudprovider/aws/aws_cloud_provider_test.go

```diff
@@ -739,3 +740,44 @@ func TestHasInstance(t *testing.T) {
 	assert.NoError(t, err)
 	assert.False(t, present)
 }
+
+// MockNodeGroup simulates the behavior of a NodeGroup for testing.
+type MockNodeGroup struct {
```

Comment:
Can we use testify.mock here instead of rolling a new mock interface by hand?  There are some other examples in this package, e.g., the `autoScalingMock` defined in `aws_wrapper_test.go`.